lib/internal/promisify.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+module.exports = promisify;
+
+function promisify(fn, single) {
+  Object.defineProperty(inner, 'name', {
+    value: fn.name + 'Promisified'
+  });
+  return inner;
+
+  function inner() {
+    if (typeof arguments[arguments.length - 1] === 'function') {
+      return fn.apply(this, arguments);
+    }
+    var resolve = null;
+    var reject = null;
+    const args = Array.from(arguments);
+    const promise = new Promise((_resolve, _reject) => {
+      resolve = _resolve;
+      reject = _reject;
```

Comment:
I don't really see why `default:` case is needed at all and doing expensive `arguments` manipulations here if `resolve` accepts just one argument anyway, and it behaves the same way whether you pass `undefined` or just don't pass any arguments.

Wouldn't this be the same? Or I'm missing something?

```javascript
function resolver(err, result) {
  return err ? reject(err) : resolve(result);
}
```